popup.js

```diff
@@ -0,0 +1,107 @@
+function getReadingListsUrlForOrigin(origin) {
+    return `${origin}/api/rest_v1/data/lists/`;
+}
+
+function readingListSetupUrlForOrigin(origin, token) {
+    return `${origin}/api/rest_v1/data/lists/setup?csrf_token=${encodeURIComponent(token)}`;
+}
+
+function readingListPostEntryUrlForOrigin(origin, listId, token) {
+    return `${origin}/api/rest_v1/data/lists/${listId}/entries/?csrf_token=${encodeURIComponent(token)}`;
+}
+
+function csrfFetchUrlForOrigin(origin) {
+    return `${origin}/w/api.php?action=query&format=json&formatversion=2&meta=tokens&type=csrf`;
+}
+
+function getCsrfToken(origin) {
+    return fetch(csrfFetchUrlForOrigin(origin), { credentials: 'same-origin' })
+    .then(res => res.json())
+    .then(res => res.query.tokens.csrftoken);
+}
+
+function getDefaultListId(url) {
+    return fetch(getReadingListsUrlForOrigin(url.origin), { credentials: 'same-origin' })
+    .then(res => {
+        if (!res.ok) {
+            return setUpReadingListsForUser(url)
+            .then(res => getDefaultListId(url));
+        }
+        return res.json()
+        .then(res => res.lists.filter(list => list.default)[0].id);
+    })
+}
+
+function setUpReadingListsForUser(url) {
+    return getCsrfToken(url.origin)
+    .then(token => fetch(readingListSetupUrlForOrigin(url.origin, token), { method: 'POST', credentials: 'same-origin' }));
+}
+
+function parseTitleFromUrl(url) {
+    return url.searchParams.has('title') ? url.searchParams.get('title') : url.pathname.replace('/wiki/', '');
```

Comment:
Hmm, yeah, I can see this having problems around the edges.  

It feels gross to do, but I guess if necessary I could use this mechanism (https://developer.chrome.com/extensions/content_scripts#host-page-communication) to grab the config variables from the `<script>` that sets them, and send them back to the extension, which would also address your other comments about preferring mw.config.*...

Reply:
That's super fragile. Getting it from `<link rel="edit">` would be still gross but at least reliable.

Reply:
done in the latest commit.

Reply:
Sorry, I haven't thought that through, it will fail when the page is not editable. rel=canonical would work I think.

Reply:
Hmm... the problem here is that `<link rel="canonical" href="https://en.wikipedia.org/wiki/%CE%92-lactam_antibiotic"/>` brings us back to the problem of having to parse the title from a full URL and not knowing which URL format is canonical for that particular wiki.

Reply:
Is applying the same parseTitleFromUrl logic as above, but to the `link[rel=canonical]` URL rather than the URL reported by the active tab, good enough for now?

Reply:
The canonical link is only present when `$wgEnableCanonicalServerLink` is true, but that's the case for all WMF wikis, so that's only a problem if you want to make the extension support third-party installations.

Otherwise, that seems like a good approach. Not sure how much difference it makes, it gets rid of the language variant in the URL at least. (ReadingLists should probably store variants; filed as [T192999](https://phabricator.wikimedia.org/T192999). Filed [T193000](https://phabricator.wikimedia.org/T193000) about the general issue of providing a reliable way to identify the article.)